src/eva/data/mon_data_space.py

```diff
@@ -689,56 +699,51 @@ def load_dset(self, vars, nvars, coords, darr, dims, ndims_used,
 
     # ----------------------------------------------------------------------------------------------
 
-    def loadChanItems(self, dataset, chan_assim, chan_nassim, channo):
-
-        """
-        Add channel as a variable using single dimension as well as chan_assim, chan_nassim, and
-        chan_yaxis_* to allow plotting of channel markers.
-
-        Args:
-            dataset (xarray.Dataset): Dataset to which variables will be added.
-            chan_assim (list): List of assimilated channel numbers.
-            chan_nassim (list): List of non-assimilated channel numbers.
-            channo (list): List of channel numbers.
-
-        Returns:
-            xarray.Dataset: Dataset with added channel-related variables.
-        """
-
-        dataset['channel'] = (['Channel'], channo)
-        if len(chan_assim) > 0:
-            dataset['chan_assim'] = (['Channel'], chan_assim)
-        if len(chan_nassim) > 0:
-            dataset['chan_nassim'] = (['Channel'], chan_nassim)
-
-        dataset['chan_yaxis_100'] = (['Channel'], [-100]*len(channo))
-        dataset['chan_yaxis_1p5'] = (['Channel'], [-1.5]*len(channo))
-        dataset['chan_yaxis_p05'] = (['Channel'], [-0.05]*len(channo))
-
-        return dataset
-
-    # ----------------------------------------------------------------------------------------------
-
-    def loadScanItems(self, dataset, scanpo):
+    def loadConditionalItems(self, dataset, chans_dict, levs_dict, scanpo):
 
         """
-        Add scan-related variables to the dataset.
+        Add channel, level, and scan related variables to the dataset.
 
         Args:
             dataset (xarray.Dataset): Dataset to which variables will be added.
+            chans_dict (dict): Dictionary of channel components.
+            levs_dict (dict): Dictionary of level components.
             scanpo (list): List of scan positions.
 
         Returns:
             xarray.Dataset: Dataset with added scan-related variables.
         """
 
-        nscan = dataset.dims.get('Scan')
-        nchan = dataset.dims.get('Channel')	  # 'Channel' is always present with 'Scan'
-        scan_array = np.zeros(shape=(nscan, nchan))
-
-        for x in range(nchan):
-            scan_array[:, x] = np.array([scanpo])
-            dataset['scan'] = (['Scan', 'Channel'], scan_array)
+        if chans_dict is not None:
+            dataset['channel'] = (['Channel'], chans_dict["chan_nums"])
+            if len(chans_dict["chans_assim"]) > 0:
+                dataset['chan_assim'] = (['Channel'], chans_dict["chans_assim"])
+            if len(chans_dict["chans_nassim"]) > 0:
+                dataset['chan_nassim'] = (['Channel'], chans_dict["chans_nassim"])
+
+            dataset['chan_yaxis_100'] = (['Channel'], [-100]*len(chans_dict["chan_nums"]))
```

Comment:
This is for plotting, right? Is there a way to do this on the plotting/YAML side rather than making pseudo data? If not, then this is fine.

Reply:
Yes its used as the y axis to position the channel and level markers.  I tried using a fixed y value (-1.5) in the yaml but that produced an error.  Should it be able to accept numeric values for x or y axes?

I did think of a slightly better way to handle the assimilated/non-assimilated data for channel and level markers.  I'll open a new issue to implement that.

Reply:
I guess I'm wondering if this is a 'transform' that can multiply a variable by X to get a new variable?

Reply:
Hmmm.  That's an interesting idea.  I think I'd need a single variable of correct dimensions with all zero values, and then in the yaml I could apply an arithmetic transform to make it what I need.   That's a much better way of handling it.  Thanks!

That's enough of a change that I think I'd like to take that up in a new issue.

Reply:
Let me walk part of that back. I just did a quick test and it works.  I'll make that change here.  I'll move this PR to draft and switch it back when I've got the change implemented and tested.